### ISSUE TITLE
[Question] Warning (bytecomp): the following functions are not known to be defined:

### ISSUE THREAD

**Issue by hongyi-zhao** [DogLooksGood/emacs-rime#131]
On Ubuntu 20.04, I compiled the latest git master version of emacs and try to install the emacs-rime into it. See following for more info.

```
$ emacs --version
GNU Emacs 28.0.50
```
The .emacs for my case:
```
;https://stackoverflow.com/questions/1595418/emacs-behind-http-proxy
(setq url-proxy-services
   '(("no_proxy" . "^\\(localhost\\|10\\..*\\|192\\.168\\..*\\|127\\..*\\|172\\.17\\..*\\)")
     ("http" . "127.0.0.1:8080")
     ("https" . "127.0.0.1:8080")))
     

;https://emacs.stackexchange.com/questions/50600/use-package-wont-load
;;{{{ Set up package and use-package
(require 'package)
(add-to-list 'package-archives
             '("melpa" . "https://melpa.org/packages/") t)
(package-initialize)

;; Bootstrap 'use-package'
(eval-after-load 'gnutls
  '(add-to-list 'gnutls-trustfiles "/etc/ssl/cert.pem"))
(unless (package-installed-p 'use-package)
  (package-refresh-contents)
  (package-install 'use-package))
(eval-when-compile
  (require 'use-package))
(require 'bind-key)
(setq use-package-always-ensure t)

;;}}}


;https://github.com/DogLooksGood/emacs-rime#%E6%9C%80%E5%B0%8F%E9%85%8D%E7%BD%AE 
(use-package rime
  :custom
  (default-input-method "rime"))
```

When I start emacs, I see the following info in its buffer:

```
[...]
Compiling file /home/werner/.emacs.d/elpa/rime-20201129.1748/rime.el at Thu Dec 10 14:18:09 2020
Warning (bytecomp): ‘focus-in-hook’ is an obsolete variable (as of 27.1); after-focus-change-function
Warning (bytecomp): reference to free variable ‘rime--current-input-key’ [3 times]
Warning (bytecomp): the following functions are not known to be defined:
    evil-normal-state-p, evil-visual-state-p, evil-motion-state-p,
    evil-operator-state-p, org-inside-LaTeX-fragment-p,
    org-inside-latex-macro-p, org-in-src-block-p, texmathp
Warning (bytecomp): defcustom for ‘rime-popup-properties’ fails to specify type [2 times]
Warning (bytecomp): defcustom for ‘rime-popup-style’ fails to specify type [2 times]
Warning (bytecomp): defcustom for ‘rime-posframe-properties’ fails to specify type [2 times]
Warning (bytecomp): defcustom for ‘rime-posframe-style’ fails to specify type [2 times]
Warning (bytecomp): defcustom for ‘rime-inline-predicates’ fails to specify type
Warning (bytecomp): defcustom for ‘rime-inline-predicates’ fails to specify containing group
Warning (bytecomp): defcustom for ‘rime-inline-predicates’ fails to specify type
Warning (bytecomp): defcustom for ‘rime-inline-predicates’ fails to specify containing group
Warning (bytecomp): defcustom for ‘rime-inline-ascii-holder’ fails to specify type
Warning (bytecomp): defcustom for ‘rime-inline-ascii-holder’ fails to specify containing group
Warning (bytecomp): defcustom for ‘rime-inline-ascii-holder’ fails to specify type
Warning (bytecomp): defcustom for ‘rime-inline-ascii-holder’ fails to specify containing group
Warning (bytecomp): defcustom for ‘rime-inline-ascii-trigger’ fails to specify type [2 times]
Warning (bytecomp): Unused lexical variable ‘input’
Warning (bytecomp): Unused lexical variable ‘sel-end’
Warning (bytecomp): Unused lexical variable ‘sel-start’
Warning (bytecomp): Unused lexical variable ‘cursor-pos’
Warning (bytecomp): Unused lexical variable ‘commit-text-preview’
Warning (bytecomp): Unused lexical variable ‘length’
Warning (bytecomp): Unused lexical argument ‘ignores’
Warning (bytecomp): Unused lexical variable ‘preedit’
Warning (bytecomp): Unused lexical argument ‘name’
Warning (bytecomp): reference to free variable ‘rime-active-mode-map’
Warning (bytecomp): reference to free variable ‘vterm-mode-map’ [2 times]
Warning (bytecomp): the following functions are not known to be defined:
    rime-lib-get-context, rime-lib-get-input, rime-lib-process-key,
    rime-lib-clear-composition, rime-lib-get-option,
    rime-lib-get-commit, rime-lib-set-cursor-pos,
    rime-lib-get-schema-list, rime-lib-select-schema, rime-lib-start,
    rime-lib-finalize, rime-lib-sync-user-data
```
Are they harm? Any hints for fixing/solving this problem will be highly appreciated? 

Regards,
HY

**Comment by DogLooksGood** [comment]
There are not harm, I will fix these warnings.

**Comment by DogLooksGood** [comment]
should be fixed in c797c443886aca6083737a3644e45af51f5b6760
